sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -341,6 +342,31 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
             this._readyTask = Task.WhenAll(readyTasks);
         }
 
+        private async Task InitializeProjectSettingsAsync(ProjectSettings projectSettings,
```

Comment:
This is the meat of the change right here. The Comparer* boilerplate noise is to implement `projectSettings.IsDefault`. The assumption here its that if the user passes non-default settings, we want to overwrite what is on disk already, and put out a warning (though not clear where warnings go?).

Alternatively we could throw and hard-fail when both non-default settings are passed in and there is something on disk already.

Now, avoiding IsDefault check would be nice but it seems sometimes our API creates default ProjectSettings invisibly so it's hard to be sure it was the user's intention. In the test it seems the user's intention is to use the settings from disk, not from the default ProjectSettings object constructed by the framework.